### ISSUE TITLE
Could this be used with MQTT?

### ISSUE THREAD

**Issue by opensourcegeek** [jq-rs/mles-rs#130]
I'm currently using MQTT client (implemented in Rust) to push data through to server. We're using cloud solutions as MQTT server which works fine, however I'm looking for a light weight option for use cases where cloud providers cannot be used. I could use RabbitMQ with MQTT adapter but I was wondering if it's possible to use this library with adapter for MQTT. I really like the simplistic approach this library has taken - thanks for open sourcing it.

**Comment by jq-rs** [comment]
Sure, one could do an MQTT/Mles proxy that would be able to pass traffic back and forth. An Mles WebSocket proxy client example already exists, so adding a new client with MQTT support using some Rust MQTT library is certainly possible by rewriting the WebSocket side. If you plan to do it, I'll certainly try to help in case questions arise about the Mles part.

**Comment by opensourcegeek** [comment]
Sounds good, but I think I need some time to get used to MLES client (for WS) before starting a MLES client for MQTT. Where is the best place to start - MLES client?

**Comment by jq-rs** [comment]
Yes, the Mles client is the best place to start as it has the proxy embedded.

**Comment by jq-rs** [comment]
As async/await is in changing state everyone is waiting for the more final APIs. Let me know if you have more questions. Closed for now.

**Comment by jq-rs** [comment]
As async/await is getting finalized and MQTT proxy seems to be a popular topic, reopening.

**Comment by jq-rs** [comment]
There is now also [mles-webproxy](https://github.com/jq-rs/mles-webproxy) available as a starting point for this enhancement.

**Comment by jq-rs** [comment]
MQTT - Mles v2 might be relevant still.

**Comment by jq-rs** [comment]
[mles-client](https://github.com/jq-rs/mles-client) includes a channel proxy now. It could be extended with MQTT proxy fairly easily, connecting MQTT messaging into Mles network.

**Comment by jq-rs** [comment]
MQTT proxy support in mles-client [v2.1.0](https://github.com/jq-rs/mles-client/releases/tag/v2.1.0)
